src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -666,3 +666,7 @@ func statefulSetKind() string {
 	var statefulSet appsv1.StatefulSet
 	return statefulSet.Kind
 }
+
+func (r *StatefulSetResource) fullConfiguratorImage() string {
```

Comment:
this whole operator-configurator relationship seems really problematic, see my comment in slack:

`this whole relationship is problematic overall, right? Let's say I am running operator 21.1.1 in production but there's already operator 21.4.4 released, but that means that I am actually running configurator 21.4.4 (latest) and operator 21.1.1 and this can change between various starts of redpanda cluster `

Reply:
If the operator starts without specifying a configurator version, then yes, "latest" would be used and if the operator is restarted the latest image would be pulled - whatever version that might be. However, the configurator tag can be specified during the operator's startup to avoid this. Perhaps the tag should be part of the cluster CR, but does this explanation make sense?